conf/cmake/ConfigureDependencies.cmake

```diff
@@ -3,167 +3,111 @@
 
+
 if (hpGEM_USE_PETSC OR hpGEM_USE_COMPLEX_PETSC)
-    set(PETSC_ARCH "" CACHE STRING "PETSc arch string should be set be the user; he will know this if he has installed PETSc")
     set(hpGEM_LOGLEVEL "DEFAULT" CACHE STRING "Verbosity of hpGEM. Default is recommended.")
 
     if (hpGEM_USE_PETSC AND hpGEM_USE_COMPLEX_PETSC)
         message(FATAL_ERROR "The options 'hpGEM_USE_PETSC' and 'hpGEM_USE_COMPLEX_PETSC' are mutually exclusive, please disable one of them")
     endif()
 
-    enable_language(C)
-    find_package(PETSc REQUIRED COMPONENTS CXX)
-
-    # Check whether we found a petsc with complex or real numbers
-    include(CheckSymbolExists)
-    check_symbol_exists(PETSC_USE_COMPLEX petscconf.h PETSC_IS_COMPLEX)
 
+    FIND_PACKAGE(PETSc REQUIRED)
     set(hpGEM_USE_ANY_PETSC ON)
 
     # Check if this matches with what the user expects
-    if (PETSC_IS_COMPLEX)
+    if (PETSc_IS_COMPLEX)
         if (hpGEM_USE_PETSC)
             message(FATAL_ERROR "Requested PETSc with real numbers, found one with complex numbers")
         endif()
-        # TODO: This should be moved
         add_definitions(-DHPGEM_USE_COMPLEX_PETSC -DHPGEM_USE_ANY_PETSC)
-
     else()
         if (hpGEM_USE_COMPLEX_PETSC)
             message(FATAL_ERROR "Requested PETSc with complex numbers, found one with real numbers")
         endif()
         add_definitions(-DHPGEM_USE_PETSC -DHPGEM_USE_ANY_PETSC)
     endif()
-    # Create target for easy linking
-    if(PETSC_FOUND)
-        add_library(PETSc::PETSc INTERFACE IMPORTED)
-        target_include_directories(PETSc::PETSc INTERFACE "${PETSC_INCLUDES}")
-        target_link_libraries(PETSc::PETSc INTERFACE "${PETSC_LIBRARIES}")
-        if(PETSC_IS_COMPLEX)
-            target_compile_definitions(PETSc::PETSc INTERFACE -DHPGEM_USE_COMPLEX_PETSC -DHPGEM_USE_ANY_PETSC)
-        else()
-            target_compile_definitions(PETSc::PETSc INTERFACE -DHPGEM_USE_PETSC -DHPGEM_USE_ANY_PETSC)
-        endif()
-
-    endif()
 
 else()
     set(hpGEM_USE_ANY_PETSC OFF)
 endif() # Petsc
 
 if(hpGEM_USE_SLEPC)
-    enable_language(C)
-    if(NOT hpGEM_USE_PETSC AND NOT hpGEM_USE_COMPLEX_PETSC)
+
+    if(NOT hpGEM_USE_ANY_PETSC)
         message(FATAL_ERROR
                 "SLEPc depends on PETSc, please also enable PETSc support")
     endif()
+
     FIND_PACKAGE(SLEPc REQUIRED)
-    add_definitions(-DHPGEM_USE_SLEPC)
-    if(NOT SLEPC_FOUND)
-        message(FATAL_ERROR
-                "The option you have chosen requires SLEPc and you do not have this installed. Please install")
-    endif()
-    # Create target for easy linking
-    if(SLEPC_FOUND)
-        add_library(SLEPc::SLEPc INTERFACE IMPORTED)
-        target_include_directories(SLEPc::SLEPc INTERFACE "${SLEPC_INCLUDE_DIRS}")
-        target_link_libraries(SLEPc::SLEPc INTERFACE "${SLEPC_LIBRARIES}")
+   
+    if(NOT(SLEPc_ARCH STREQUAL PETSc_ARCH))
+        message(FATAL_ERROR "SLEPc and PETSc architecture differ. ${SLEPc_ARCH} vs ${PETSc_ARCH}")
```

Comment:
I get an error here:
```
CMake Error at conf/cmake/ConfigureDependencies.cmake:95 (message):
  SLEPc and PETSc architecture differ.  #define SLEPC_PETSC_ARCH "" vs
  #define PETSC_ARCH ""
```
Might it be that the parsing is done incorrectly and that `PETSC_ARCH` variant contains a newline at the start.

Reply:
yeah I think the regex fails on empty ARCH